SProkopios/enhancement/236-creating-MultiSelectionDropdown

SProkopios opened the pull request:
## 📄 **Pull Request Overview**

**Issue Number**: #256

## 🔧 **Changes Made**

1. Created a MultiSelectionDropdown component within CustomForm. Created NewClanForm.sotries.tsx.

2. [Any refactoring or clean-up tasks]

---

## ✅ **Checklist Before Submission**

- **Functionality**: I have tested my code, and it works as expected. ✅
- **JSDoc**: I have added or updated JSDoc comments for all relevant code. ✅
- **Debugging**: No `console.log()` or other debugging statements are left. ✅
- **Clean Code**: Removed commented-out or unnecessary code. ✅
- **Tests**: Added new tests or updated existing ones for the changes made. ✅ *tested with storybook
- **Documentation**: Documentation has been updated (if applicable).

---

## 📝 **Additional Information**

Provide any additional context or information that reviewers may need to know:

- **Screenshots**: [Include any screenshots or videos if the changes affect the UI]
- **Dependencies**: react-multi-select-component

- **Known Issues**: Not yet

Skoivumaki commented:
Your branch is failing to build since you are trying to import client component code inside a server component. I fixed the issue and tried to use your example code which didnt work. `@example <Form.MultiSelectionDropdown key={'tag'} maxSelections={3} label={'Tagi'} options={clans} value={selected} onSelectChange={(newSelection) => setSelected(selection)} />` (you need to explain how onSelectChange is expected to be used also, why the "tag" stuff)

> *tested with storybook

How?

I pushed enums that are expected to work with this multiselection dropdown. (see commits/files changed)
**Todo:**

- [ ] Please test using `ClanLabel` enums and screenshot results of it working in `NewClanForm.tsx` etc. (see screenshot)
- [ ] When it works update tests if needed
- [ ] Update JSDoc `@example` to something functional with placeholder data (see InputField for example etc)

Example of what multiselection in NewClanForm could look like. I had to map enums first and used onSelectChange according to [modules documentation](https://github.com/hc-oss/react-multi-select-component#readme):
![image](https://github.com/user-attachments/assets/de55bd30-1d42-4024-bcf8-a4826a400101)

SProkopios commented:
Todo:

Please test using ClanLabel enums and screenshot results of it working in NewClanForm.tsx etc. (see screenshot) ✅  *Added story to "CustomForm.stories.tsx".
When it works update tests if needed ✅ 
Update JSDoc @example to something functional with placeholder data (see InputField for example etc) ✅ 
Make sure branch builds and tests succeed  ✅ *Caution: Added 'use client' to CustomForm.tsx for pass the build!

Added Story to  "CustomForm.stories.tsx" to test and display how to code works. Also refactored options to accept ClanLabel enums!

Here is a pic from NewClanForm.tsx:
![Multiselection1](https://github.com/user-attachments/assets/8422a954-c027-4d9e-bf4e-5711fa201144)

Skoivumaki commented on frontend-next-migration/src/shared/ui/CustomForm/ui/CustomForm.tsx:
```diff
@@ -1,3 +1,4 @@
+'use client';
```
> *Caution: Added 'use client' to CustomForm.tsx for pass the build!

You could have used simple if else magic instead of `useState` & `useEffect` to keep CustomForm.tsx server sided. But since whatever CustomForm is connected to is most likely a client component, it wouldnt really matter anyway. Its just needed to pass the build really, which I understand. (if client side component existed and used this dropdown, build would pass even without declaring `use client` in `CustomForm.tsx`)

Skoivumaki commented on frontend-next-migration/src/shared/ui/CustomForm/ui/CustomForm.tsx:
```diff
@@ -132,13 +136,107 @@ function Button({ children, className = '', ...props }: ButtonProps) {
 }
 
+/**
+ * MultiSelectionDropdown component for rendering a multi-selection dropdown.
```
Example code works. Personally I would have added more props that `MultiSelectionDropdown` uses, like `disableSearch`.

Skoivumaki reviewed (approved):
I tested it in new clan form and it works, nice.
